runtime/server/auth/claims.go

```diff
@@ -0,0 +1,76 @@
+package auth
+
+import (
+	"github.com/golang-jwt/jwt/v4"
+)
+
+// Claims resolves permissions for a requester.
+type Claims interface {
+	// Subject returns the token subject if present (usually a user or service ID)
+	Subject() string
+	// Can resolves system-level permissions.
+	Can(p Permission) bool
+	// CanInstance resolves instance-level permissions.
+	CanInstance(instanceID string, p Permission) bool
+}
+
+// jwtClaims implements Claims and resolve permissions based on a JWT payload.
+type jwtClaims struct {
+	jwt.RegisteredClaims
+	System    []Permission            `json:"sys,omitempty"`
+	Instances map[string][]Permission `json:"ins,omitempty"`
+}
+
+func (c *jwtClaims) Subject() string {
+	return c.RegisteredClaims.Subject
+}
+
+func (c *jwtClaims) Can(p Permission) bool {
+	for _, p2 := range c.System {
+		if p2 == p {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *jwtClaims) CanInstance(instanceID string, p Permission) bool {
+	for _, p2 := range c.Instances[instanceID] {
+		if p2 == p {
+			return true
+		}
+	}
+	return c.Can(p)
```

Comment:
Yeah, this is intentional, but wasn't quite sure if it was a good idea. My thinking was that this would make it easy to make admin tokens that could do anything. But maybe it's an anti-pattern – maybe we should require instance-level actions to be scoped by an instance ID?

Might just keep it this way for now as we can make it more restrictive in the future. Just have to be careful what kinds of tokens we issue.